### PR TITLE
Render animated sprites in the hail panel correctly

### PR DESCRIPTION
**Pull request by warp-core**
**Bug fix**

This PR addresses the bug/feature described in issue #9766

## Summary
Ships and planets in the hail panel should now be properly animated.

## Testing Done
No.

### CODE REVIEW

**Comment by tibetiroka** [comment]
Seems to work for both planets and ships.

File: source/HailPanel.h [hunk 1]
```diff
@@ -58,9 +58,11 @@ class HailPanel : public Panel {
 	PlayerInfo &player;
 	std::shared_ptr<Ship> ship = nullptr;
 	std::function<void(const Government *)> bribeCallback = nullptr;
+	const StellarObject *object = nullptr;
 	const Planet *planet = nullptr;
 	const Sprite *sprite = nullptr;
```
Comment (RisingLeaf) [inline source/HailPanel.h:63]: we dont need this anymore, do we?

Comment (warp-core) [reply source/HailPanel.h:63]: Removing them would require a more significant refactor than I think is necessary to fix the bug. I might make some larger changes later. Leaving them here doesn't really cost anything (performance isn't a particularly high priority in the hail panel) and provides more convenient access on member methods.

Comment (RisingLeaf) [reply source/HailPanel.h:63]: sprite is only used in one other occasion, I am only referring to sprite, not to planet

Comment (warp-core) [reply source/HailPanel.h]: Looks like you're right. I've removed HailPanel::sprite.

**Pull request merged**